src/Microsoft.OData.Core/UriParser/TreeNodeKinds/QueryNodeKind.cs

```diff
@@ -158,6 +158,11 @@ public enum QueryNodeKind
         /// Count of a collection contains primitive or enum or complex or entity type.
         /// </summary>
         Count = InternalQueryNodeKind.Count,
+
+        /// <summary>
+        /// Cast on a single property.
+        /// </summary>
+        SinglePropertyCast = InternalQueryNodeKind.SinglePropertyCast,

```

Comment:
Resource / Property makes more sense than Resource / Primitive.

---
In reply to: [124892629](https://github.com/OData/odata.net/pull/848#discussion_r124892629) [](ancestors = 124892629)